              ->bind(':link_id', $link['id'], ParameterType::INTEGER);
+                    $this->db->setQuery($linkUpdate)->execute();
+                    $linksUpdated++;
+                }
+            }
+
+            $this->db->transactionCommit();
+            $started = false;
+        } catch (Throwable $e) {
+            if ($started) {
+                try {
+                    $this->db->transactionRollback();
+                } catch (Throwable) {
+                }
+            }
+            throw $e;
+        }
+
+        return ['cards_updated' => $cardsUpdated, 'links_updated' => $linksUpdated];
+    }
+
     public function getLinkedCompetitionSeasonId(int $cardId): int
     {
         if ($cardId < 1) {
@@ -521,6 +690,22 @@
         $this->db->setQuery($query)->execute();
     }
 
+    private function normalizeSyncDate(?string $value, string $field): ?string
+    {
+        $value = $value !== null ? trim($value) : '';
+        if ($value === '') {
+            return null;
+        }
+
+        $date = \DateTimeImmutable::createFromFormat('!Y-m-d', $value);
+        $errors = \DateTimeImmutable::getLastErrors();
+        if (!$date || ($errors !== false && (($errors['warning_count'] ?? 0) > 0 || ($errors['error_count'] ?? 0) > 0)) || $date->format('Y-m-d') !== $value) {
+            throw new \InvalidArgumentException('Invalid competition season date for ' . $field . '.');
+        }
+
+        return $value;
+    }
+
     private function relationCondition(string $alias = ''): string
     {
         $column = $alias !== '' ? $alias . '.relation_type' : 'relation_type';
--- a/component/admin/src/View/Cardbulk/HtmlView.php
+++ b/component/admin/src/View/Cardbulk/HtmlView.php
@@ -55,7 +55,7 @@
 
         AdminAssetService::useAssets($this->getDocument());
         $base = rtrim(Uri::root(true), '/') . '/media/com_decaromembership';
-        $this->getDocument()->addScript($base . '/js/cardbulk.js', ['version' => '1.9.31'], ['defer' => true]);
+        $this->getDocument()->addScript($base . '/js/cardbulk.js', ['version' => '1.9.32'], ['defer' => true]);
 
         ToolbarHelper::title(Text::_('COM_DECAROMEMBERSHIP_CARDBULK_TITLE'), 'copy');
         ToolbarHelper::link('index.php?option=com_decaromembership&view=records&entity=cards', Text::_('JTOOLBAR_CLOSE'), 'arrow-left');
--- a/component/decaromembership.xml
+++ b/component/decaromembership.xml
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <extension type="component" method="upgrade">
-  <name>COM_DECAROMEMBERSHIP</name><author>Luca De Caro</author><creationDate>2026-09</creationDate><copyright>Copyright (C) 2026 Luca De Caro. All rights reserved.</copyright><license>GNU General Public License version 2 or later</license><authorUrl>https://github.com/xdecaro/membership</authorUrl><version>1.9.31</version><description>COM_DECAROMEMBERSHIP_XML_DESCRIPTION</description><targetplatform name="joomla" version="6.*" /><namespace path="src">Xdecaro\Component\Decaromembership</namespace>
+  <name>COM_DECAROMEMBERSHIP</name><author>Luca De Caro</author><creationDate>2026-09</creationDate><copyright>Copyright (C) 2026 Luca De Caro. All rights reserved.</copyright><license>GNU General Public License version 2 or later</license><authorUrl>https://github.com/xdecaro/membership</authorUrl><version>1.9.32</version><description>COM_DECAROMEMBERSHIP_XML_DESCRIPTION</description><targetplatform name="joomla" version="6.*" /><namespace path="src">Xdecaro\Component\Decaromembership</namespace>
   <administration><menu>COM_DECAROMEMBERSHIP</menu><submenu><menu view="dashboard">COM_DECAROMEMBERSHIP_DASHBOARD</menu><menu link="option=com_decaromembership&amp;view=records&amp;entity=members">COM_DECAROMEMBERSHIP_MEMBERS</menu><menu link="option=com_decaromembership&amp;view=records&amp;entity=categories">COM_DECAROMEMBERSHIP_CATEGORIES</menu><menu link="option=com_decaromembership&amp;view=records&amp;entity=cases">COM_DECAROMEMBERSHIP_CASES</menu><menu link="option=com_decaromembership&amp;view=records&amp;entity=renewals">COM_DECAROMEMBERSHIP_RENEWALS</menu><menu link="option=com_decaromembership&amp;view=records&amp;entity=dues">COM_DECAROMEMBERSHIP_DUES</menu><menu link="option=com_decaromembership&amp;view=records&amp;entity=cards">COM_DECAROMEMBERSHIP_CARDS</menu><menu link="option=com_decaromembership&amp;view=records&amp;entity=card_numbering_rules">COM_DECAROMEMBERSHIP_CARD_NUMBERING_RULES</menu><menu link="option=com_decaromembership&amp;view=records&amp;entity=payments">COM_DECAROMEMBERSHIP_PAYMENTS</menu><menu link="option=com_decaromembership&amp;view=records&amp;entity=transfers">COM_DECAROMEMBERSHIP_TRANSFERS</menu><menu view="information">COM_DECAROMEMBERSHIP_INFORMATION</menu></submenu><files folder="admin"><filename>access.xml</filename><filename>config.xml</filename><folder>services</folder><folder>sql</folder><folder>src</folder><folder>tmpl</folder></files><languages folder="admin/language"><language tag="en-GB">en-GB/com_decaromembership.ini</language><language tag="en-GB">en-GB/com_decaromembership.sys.ini</language><language tag="fr-FR">fr-FR/com_decaromembership.ini</language><language tag="fr-FR">fr-FR/com_decaromembership.sys.ini</language><language tag="it-IT">it-IT/com_decaromembership.ini</language><language tag="it-IT">it-IT/com_decaromembership.sys.ini</language></languages></administration>
   <media destination="com_decaromembership" folder="media"><filename>joomla.asset.json</filename><folder>css</folder><folder>js</folder></media>
   <install><sql><file driver="mysql" charset="utf8">sql/install.mysql.utf8mb4.sql</file></sql></install><uninstall><sql><file driver="mysql" charset="utf8">sql/uninstall.mysql.utf8mb4.sql</file></sql></uninstall><update><schemas><schemapath type="mysql">sql/updates/mysql</schemapath></schemas></update>
--- a/component/media/joomla.asset.json
+++ b/component/media/joomla.asset.json
@@ -1 +1 @@
-{"$schema":"https://developer.joomla.org/schemas/json-schema/web_assets.json","name":"com_decaromembership","version":"1.9.31","description":"Membership by xdecaro assets","license":"GPL-2.0-or-later","assets":[{"name":"com_decaromembership.admin","type":"style","uri":"com_decaromembership/css/admin.css","version":"1.9.31"},{"name":"com_decaromembership.core-bridge","type":"style","uri":"com_decaromembership/css/core-bridge.css","version":"1.9.31","dependencies":["com_decaromembership.admin"]},{"name":"com_decaromembership.admin","type":"script","uri":"com_decaromembership/js/admin.js","version":"1.9.31","dependencies":["core"]}]}
+{"$schema":"https://developer.joomla.org/schemas/json-schema/web_assets.json","name":"com_decaromembership","version":"1.9.32","description":"Membership by xdecaro assets","license":"GPL-2.0-or-later","assets":[{"name":"com_decaromembership.admin","type":"style","uri":"com_decaromembership/css/admin.css","version":"1.9.32"},{"name":"com_decaromembership.core-bridge","type":"style","uri":"com_decaromembership/css/core-bridge.css","version":"1.9.32","dependencies":["com_decaromembership.admin"]},{"name":"com_decaromembership.admin","type":"script","uri":"com_decaromembership/js/admin.js","version":"1.9.32","dependencies":["core"]}]}--- a/package/pkg_decaromembership.xml
+++ b/package/pkg_decaromembership.xml
@@ -1 +1 @@
-<?xml version="1.0" encoding="utf-8"?><extension type="package" method="upgrade"><name>Membership by xdecaro</name><packagename>decaromembership</packagename><version>1.9.31</version><creationDate>2026-09</creationDate><author>Luca De Caro</author><authorUrl>https://github.com/xdecaro/membership</authorUrl><copyright>Copyright (C) 2026 Luca De Caro. All rights reserved.</copyright><license>GNU General Public License version 2 or later</license><description>Membership management for Joomla.</description><targetplatform name="joomla" version="6.*"/><scriptfile>script.php</scriptfile><files><file type="component" id="com_decaromembership">com_decaromembership.zip</file><file type="plugin" id="decaromembership" group="xdecaroanalytics">plg_xdecaroanalytics_decaromembership.zip</file><file type="plugin" id="decaromembership" group="task">plg_task_decaromembership.zip</file></files><updateservers><server type="extension" priority="1" name="Membership by xdecaro">https://raw.githubusercontent.com/xdecaro/membership/main/updates/pkg_decaromembership.xml</server></updateservers><changelogurl>https://raw.githubusercontent.com/xdecaro/membership/main/updates/changelog.xml</changelogurl></extension>
+<?xml version="1.0" encoding="utf-8"?><extension type="package" method="upgrade"><name>Membership by xdecaro</name><packagename>decaromembership</packagename><version>1.9.32</version><creationDate>2026-09</creationDate><author>Luca De Caro</author><authorUrl>https://github.com/xdecaro/membership</authorUrl><copyright>Copyright (C) 2026 Luca De Caro. All rights reserved.</copyright><license>GNU General Public License version 2 or later</license><description>Membership management for Joomla